src/Microsoft.DotNet.Watcher.Core/Internal/ProjectWatcher.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.Watcher.Core.Internal
+{
+    public class ProjectWatcher : IDisposable
+    {
+        private readonly IProjectProvider _projectProvider;
+        private readonly IFileWatcher _fileWatcher;
+
+        private readonly string _rootProject;
+        private readonly bool _watchProjectJsonOnly;
+
+        private ISet<string> _watchedFiles;
+
+        public ProjectWatcher(
+            string projectToWatch,
+            bool watchProjectJsonOnly,
+            Func<IFileWatcher> fileWatcherFactory,
```

Comment:
I don't see the point of passing in the factory instead of just the `IFileWatcher` if all we are going to do is just create an instance in the constructor.